var/spack/repos/builtin/packages/py-netpyne/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyNetpyne(PythonPackage):
+    """Netpyne: A python package to facilitate the development,
+    parallel simulation, optimization and analysis of multiscale
+    biological neuronal networks in NEURON."""
+
+    homepage = "http://www.netpyne.org/"
+    url = "https://github.com/suny-downstate-medical-center/netpyne/archive/refs/tags/v1.0.3.1.tar.gz"
+    git = "https://github.com/suny-downstate-medical-center/netpyne.git"
+
+    version("master", git=git, branch="master")
+    version("1.0.3.1", sha256="4f8492d58ff1dd7ec5ba6ed1f58f94548b8c1e4e9fd50b8a6d2e9f8eb400736d")
+
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-numpy", type="run")
+    depends_on("py-scipy", type="run")
+    depends_on("py-matplotlib@:3.5.1", type="run")
+    depends_on("py-matplotlib-scalebar", type="run")
+    depends_on("py-future", type="run")
+    depends_on("py-pandas", type="run")
+    depends_on("py-bokeh", type="run")
+    depends_on("py-schema", type="run")
+    depends_on("py-lfpykit", type="run")
+
+
+    depends_on("python@3.7")
```

Comment:
Why?

Reply:
https://github.com/suny-downstate-medical-center/netpyne/blob/v1.0.3.1/setup.py#L66 setup.py seems quite specific about their desire for 3.7 (or earlier, but you've already explained that Spack dropped support for that).

Reply:
These classifiers have no effect on concretization, it's just a list of versions the authors have personally tested at the time of release. Unless you know for a fact that newer versions don't work, I would remove this line.

Reply:
I most certainly don't know, so I've cleaned it up :-)